src/sentry/models/authprovider.py

```diff
@@ -57,6 +58,14 @@ class Meta:
     def __str__(self):
         return self.provider
 
+    @transaction.atomic()
+    def save(self, *args, **kwargs):
+        """
+        For migration to IdentityProvider.
+        Wrap in a transaction so the pre_save will dual-write to IdentityProvider
+        """
+        super().save(*args, **kwargs)
+
```

Comment:
It's worth checking whether this works when we create a new row. I don't think it will work when we call `update()`, so you'll need to put this there as well.